akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/Effect.scala

```diff
@@ -192,6 +192,11 @@ trait EffectBuilder[+Event, State] extends Effect[Event, State] {
    */
   def thenNoReply(): ReplyEffect[Event, State]
 
+  /**
+   * Persist additional metadata together with the event(s).
+   */
+  def persistMetadata(metadata: AnyRef): EffectBuilder[Event, State]
```

Comment:
One thing to note is that when persisting more than one event the same metadata will be used for all events. Good enough?

Reply:
We can extend if we decide it is needed at some point in the future.

Reply:
changed in https://github.com/akka/akka/pull/32628/commits/dfcbebc414cc2a794295e4578605e4e87da40def